### PR TITLE
[new feature] Article notes field

### PR DESCRIPTION
**Pull request by brianteeman**
We have a note field for com_categories, com_menus and com_modules
This PR adds the functionality to com_content as it has been requested several times and can be very useful

## To test
Apply the pr AND the database fix (we added a new field)
Create an article and on the right, above version note you will see a new field called note
Enter a note and save the article
You can now see the note with the article in the list
You can also search by the note
Check the featured article list
Check the article modal list (article xtd-editor button or from creating a new single article menu)

On the frontend you can also add/edit a note. You will see it on the publishing tab above the version note field

### Bonus
When adding this field to the frontend beez template I spotted that the version note field wasnt shown there either so I added that at the same time

### CODE REVIEW

File: administrator/components/com_admin/sql/updates/postgresql/3.9.0-2017-12-22.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE `#__content` ADD COLUMN `note` VARCHAR(255) NOT NULL DEFAULT '';
+
```
Comment (csthomas) [inline administrator/components/com_admin/sql/updates/postgresql/3.9.0-2017-12-22.sql]: Use '"' (double quote) instead '`'  to escape names in postgresql.

Comment (brianteeman) [reply administrator/components/com_admin/sql/updates/postgresql/3.9.0-2017-12-22.sql]: thanks - should be ok now

File: administrator/components/com_content/views/articles/tmpl/modal.php [hunk 2]
```diff
@@ -138,6 +138,13 @@
 							<a class="select-link" href="javascript:void(0)" <?php echo $attribs; ?>>
 								<?php echo $this->escape($item->title); ?>
 							</a>
+							<span class="small" title="<?php echo $this->escape($item->path); ?>">
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Same fix here with the underlined space.

File: administrator/components/com_content/models/forms/article.xml [hunk 3]
```diff
@@ -37,6 +37,16 @@
 			size="40" 
 		/>
 
+		<field
+			name="note"
+			type="text"
+			label="COM_CONTENT_FIELD_NOTE_LABEL"
+			description="COM_CONTENT_FIELD_NOTE_DESC"
+			class="span12"
```
Comment (Quy) [inline administrator/components/com_content/models/forms/article.xml:45]: change to `inputbox`

File: administrator/components/com_content/views/articles/tmpl/modal.php (hunk 2 above)
Comment (Quy) [reply administrator/components/com_content/views/articles/tmpl/modal.php]: `PHP Notice:  Undefined property: stdClass::$path in\administrator\components\com_content\views\articles\tmpl\modal.php on line 141`

File: administrator/components/com_content/models/forms/article.xml (hunk 3 above)
Comment (brianteeman) [reply administrator/components/com_content/models/forms/article.xml:45]: Please check the other notes fields in modules and menu. I am sure this si the same

File: components/com_content/models/forms/article.xml [hunk 4]
```diff
@@ -124,6 +124,16 @@
 			size="20" 
 		/>
 
+		<field
+			name="note"
+			type="text"
+			label="COM_CONTENT_FIELD_NOTE_LABEL"
+			description="COM_CONTENT_FIELD_NOTE_DESC"
+			class="span12"
```
Comment (Quy) [inline components/com_content/models/forms/article.xml]: change to `inputbox`

File: media/com_wrapper/js/iframe-height.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 function iFrameHeight()
 {
     var height = 0;
-    var iframe = document.getElementById('blockrandom');
+    var iframe = document.querySelectorAll('*[id^="blockrandom"]');
```
Comment (Quy) [inline media/com_wrapper/js/iframe-height.js]: Should not be in this PR.

File: templates/beez3/html/com_content/form/edit.php [hunk 6]
```diff
@@ -229,6 +229,22 @@
...
+					<?php echo $this->form->getInput('note'); ?>
+				</div>
+			</div>			
+			<?php if ($params->get('save_history', 0)) : ?>
+				<div class="control-label">
+					<?php echo $this->form->getLabel('version_note'); ?>
+				</div>
+				<div class="controls">
+					<?php echo $this->form->getInput('version_note'); ?>
+				</div
```
Comment (Quy) [inline templates/beez3/html/com_content/form/edit.php]: Add closing `>`

**Comment by Quy** [comment]
On the frontend, the note field is wider due to the `span12` class.

note-frontend

File: administrator/components/com_content/views/articles/tmpl/modal.php [hunk 7]
```diff
@@ -136,8 +136,14 @@
 								. ' data-language="' . $this->escape($lang) . '"';
 							?>
 							<a class="select-link" href="javascript:void(0)" <?php echo $attribs; ?>>
-								<?php echo $this->escape($item->title); ?>
-							</a>
+								<?php echo $this->escape($item->title); ?></a>
+							<span class="small">
```
Comment (Quy) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Add `break-word`?

Comment (brianteeman) [reply administrator/components/com_content/views/articles/tmpl/modal.php]: might as well - wont do any harm

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c78fdf42a375be2187bb3cdc383f34bc02d72de1<br /><br />Tested with MySQL<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19134">issues.joomla.org/tracker/joomla-cms/19134</a>.</sub>

**Review by alikon — CHANGES_REQUESTED** [review]
the update is wrong you need a single quote before the 
`{"common....`. and another one at the end `fulltext"}}`

should be something like 
`
'{"common":{"core_content_item_id":"id","core_title":"title","core_state":"state","core_alias":"alias","core_created_time":"created","core_modified_time":"modified","core_body":"introtext", "core_hits":"hits","core_publish_up":"publish_up","core_publish_down":"publish_down","core_access":"access", "core_params":"attribs", "core_featured":"featured", "core_metadata":"metadata", "core_language":"language", "core_images":"images", "core_urls":"urls", "core_version":"version", "core_ordering":"ordering", "core_metakey":"metakey", "core_metadesc":"metadesc", "core_catid":"catid", "core_xreference":"xreference", "asset_id":"asset_id", "note":"note"}, "special":{"fulltext":"fulltext"}}'`

the same fix on postgresql and mssql

**Comment by brianteeman** [comment]
@alikon all done - thanks

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on e67b6035331f09f8c7d7ac1021485f24a8eaa39f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19134">issues.joomla.org/tracker/joomla-cms/19134</a>.</sub>

**Review by alikon — APPROVED** [review]

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e67b6035331f09f8c7d7ac1021485f24a8eaa39f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19134">issues.joomla.org/tracker/joomla-cms/19134</a>.</sub>

**Comment by Quy** [comment]
RTC

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19134">issues.joomla.org/tracker/joomla-cms/19134</a>.</sub>

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**